프로젝트 리뷰 부탁드립니다.

102092 opened the pull request:
안녕하세요 !!
지금까지 구현한 것들 리뷰 부탁드립니다.
많은 조언 부탁드립니다 🙇 

### 구현 목표
- Codesquad 멤버들을 위한 Velog 같은 블로그 구현

### 사용 기술
- Spring Data JPA, Spring Rest Docs, MySQL...

102092 committed:
[#1] Chore : 프로젝트 생성

기본 의존성을 추가하여, 프로젝트를 생성함

102092 committed:
Chore : gitignore 수정

102092 committed:
Chore : docker-compose 설정

local에서 사용할 docker에 대한 설정 추가
mysql: 5.7.29, root password, character-set, collation..

102092 committed:
[#3] Chore : test시에만 H2를 사용하도록 변경

102092 committed:
[#3] Chore : local application.yml 설정

- datasource, loggin, actuator 기본적인 설정만
- 추가적으로 변경 예정

102092 committed:
[#3] Chore : test application.yml 설정

102092 committed:
[#3] Chore : actuator 설정 추가

- base-path /system/actuator
- health chceck option 추가

Hanul-Choi committed:
[#4] Feat : Github Actions CI 업로드

- 수월환 자동 배포 환경을 만들기 위해 CI/CD 만드는 중

Hanul-Choi committed:
[#4] Feat : AWS CodeDeploy 에 사용할 appspec.yml 업로드

- 수월환 자동 배포 환경을 만들기 위해 CI/CD 만드는 중.

Hanul-Choi committed:
[#4] Feat : deploy.sh 업로드

- 수월환 자동 배포 환경을 만들기 위해 CI/CD 만드는 중.
- appspec.yml 에서 deploy.sh의 위치를 지정해주면 배포할 때 deploy.sh가 실행되어 자동 배포처럼 보이게 해준다.

Hanul-Choi committed:
[#4] Modify : 주석 내용 수정

Hanul-Choi committed:
[#4] Feat : Github Actions CD 업로드

- 수월환 자동 배포 환경을 만들기 위해 CI/CD 만드는 중

hanurii committed:
Update issue templates

잦은 이슈 생성에 따른 이슈 템플릿 생성

hanurii committed:
Update custom.md

102092 committed:
Merge pull request #14 from Hydronium-ion/feature/issue-004-be

서버 CI 기능 추가

102092 committed:
Merge pull request #15 from Hydronium-ion/feature/issue-004-be-CD

서버 CD 기능 추가

Commit:
[#4] Feat : application.yml에 prod 설정 추가

- 배포 시 prod 설정을 가지고 배포를 진행할 수 있게하기 위해서 추가

102092 committed:
Revert "서버 CD 기능 추가"

102092 committed:
Merge pull request #17 from Hydronium-ion/revert-15-feature/issue-004-be-CD

CD 기능 추가

Commit:
[#3] Chore :  현재 사용하지 않는 prod 설정 내 flyway 항목 삭제

- 추후 사용할 수 있지만 지금은 의존성을 추가하지 않았으므로 사용하지 않음

Commit:
[#4] Feat : Github Actions CD 업로드

- 수월환 자동 배포 환경을 만들기 위해 CI/CD 만드는 중

Commit:
[#4] Modify : 필요없는 설정 삭제

- working directory가 BE로 되어 있어서 삭제

hanurii committed:
Merge pull request #18 from Hydronium-ion/feature/issue-004-be

서버 CD 기능 추가2

102092 committed:
Merge pull request #21 from Hydronium-ion/feature/issue-003

프로젝트 설정

102092 committed:
[#4] Fix : CD master branch에 push 되었을 경우에 동작하도록

102092 committed:
Merge pull request #22 from Hydronium-ion/feature/issue-004

[#4] Fix : CD master branch에 push 되었을 경우에 동작하도록

102092 committed:
Merge pull request #23 from Hydronium-ion/dev

1주차 프로젝트 환경 설정

hanurii committed:
[#4] Chore : code-deploy 수정

- 같은 AWS code-deploy 를 사용하면 한 쪽에서 배포를 했을 때(예:Client) 다른 쪽(예:Server) 의 디렉토리가 사라지는 현상이 발생했다.
- 그 현상을 고치고자 따로 code-deploy 를 만들었다.

hanurii committed:
Merge pull request #24 from Hydronium-ion/feature/issue-004-updatecodedeploy

AWS CodeDeploy 애플리케이션 수정

hanurii committed:
Merge pull request #25 from Hydronium-ion/dev

AWS CodeDeploy 자동 배포 시 Client(or Server) 디렉토리 사라지는 현상 해결 Test

102092 committed:
[#7] Chore : prod actuator 설정 추가

102092 committed:
[#7] Feat : Main Page Domain 생성

- Account User 역할, syntax 문제로 변경
- Like : 좋아요, 여러 유저는 여러개의 좋아요를 만들 수 있음. Relation 테이블 필요함

102092 committed:
[#7] Chore : ddl-auto 옵션 수정

- create -> create-drop

102092 committed:
[#7] Refactor : Entity 연관관계 제거

- 빠른 CRUD 기능 구현을 위해, 우선 연관관계 제거
- 기능 구현 후 추후 매핑 예정

102092 committed:
[#11] Feat : JackSon 설정 추가

- 기본 DateTimeFormatting 설정

102092 committed:
[#11] Feat : Main 페이지 조회를 위한 Post Model 작성

- MVC 구조를 위한 Service, Controller 작성
- Application Runner를 상속한 데이터 추가

102092 committed:
[#11] Feat : BaseTimeEntity 생성

- 생성, 시간을 관리할 추상 객체
- 현재는 Post만 상속하고 있음.

102092 committed:
[#11] Feat : ApiResult, Error 객체 생성

- Api 응답에 사용할 객체, Error 에 사용할 기본 객체를 생성함
- 추후 enum등을 이용해서 Error code를 핸들링할 생각

102092 committed:
[#11] Feat : RestDocs 적용

- Post FindAll() 메서드 에 대한 테스트 작성
- RestDocs 적용
- 추후 Docs의 template을 보기 좋게 수정할 예정

102092 committed:
[#11] Refactor : ApiError 접근제어자 수정

102092 committed:
[#11] Feat : 기본 Error Handle 설정

- Entity NotFound에 대한 에러 처리
- 추후, 인증등의 에러 핸들링 추가 예정

102092 committed:
[#11] Feat : Post Id로 조회 하는 기능 추가

- 조회 중 오류 발생 시(해당하는 Post를 못찾았을 경우), NotFoundException 발생

102092 committed:
[#11] Refactor : CommandLiner 수정

- DB와 연결되는 회수를 1번으로 수정

102092 committed:
[#11] Feat : 메인페이지 최신순 정렬 기능

- 기본적으로 생성시간 내림차순으로 진행됨
- 이를 위해 Post Entity에서 @CreationTimeStamp를 주석 처리함

102092 committed:
[#11] Feat : api-guide.adoc 포멧 수정

- table content 추가
- 필요한 snippets만 추가함

102092 committed:
[#11] Refactor : @Transactional 수정

- 코드 중복을 막기 위해, 클래스 위에 @Transactional(...) 작성
- readonly가 필요하지 않는 곳에는 @Transactional 명시

102092 committed:
[#11] Refactor : 사용하지 않는 메서드 삭제, ApiError 생성자의 접근제어자 수정

- ExceptionHandler에서 사용하지 않을 newResponse 메서드 삭제
- ApiError 생성자의 접근제어자 수정

102092 committed:
[#11] Refactor : 사용하지 않는 변수 삭제

- days 변수 삭제
- 코드 정렬

hanurii committed:
Merge pull request #26 from Hydronium-ion/feature/issue-011

Post 엔티티, Spring Rest Docs 기본 설정, Post 메인페이지(최신순으로 정렬기능)

102092 committed:
Merge pull request #28 from Hydronium-ion/dev

1차 배포

102092 committed:
[#29] Fix : application.yml 수정

- ddl auto validate -> create 로 수정

102092 committed:
Merge pull request #30 from Hydronium-ion/fix

[#29] Fix : application.yml 수정

102092 committed:
Merge pull request #31 from Hydronium-ion/fix

Fix

hanurii committed:
[#33] Feat : Post 생성,삭제 기능 추가

- 간단하게 생성,삭제 기능을 만들어보았습니다.
- request body를 받을 DTO를 생성했습니다.
- 지금은 JpaRepository 를 상속한 PostRepository 를 이용해 save, delete를 했지만 추후엔 엔티티매니저를 사용하여 구현해보고 싶습니다.

hanurii committed:
[#33] Feat : 포스트 삭제 시 getOne() 메서드가 제대로 exception을 throw 하지 못하는 문제로 메서드 변경

- 포스트 삭제 시 getOne() 메서드가 제대로 exception을 throw 하지 못하는 문제로 메서드 변경
- getOne() 메서드보다는 findById() 를 사용하는 방향으로 하는 것이 좋을 것 같다

hanurii committed:
Feat : p6spy 의존성 추가

- Sprng Data JPA 로 구동 시 쿼리문에 ? 가 나오면 어떤 값인지 확인 할 방법이 없음
- p6spy 의존성을 추가하면 어떤 값이 들어가는지 로그로 확인 할 수 있어서 편리함

hanurii committed:
[#33] Feat : 포스트 생성 테스트 코드 작성

- 포스트가 생성되는지 확인하는 메서드 작성

hanurii committed:
[#33] Feat : PostService.Create() 수정과 PostServiceTest 수정

- PostService.Create()의 파라미터로 PostRequestDto를 받는데 이 DTO를 Post 객체 생성하는 걸로 만드려니 Service 클래스가 너무 길어져서 Post 엔티티 안으로 이동시킴
- PostServiceTest.create_tenTimes_AllPass() 리팩토링

hanurii committed:
[#33] Feat : delete() 리턴값 Post 객체 반환

- 테스트 검증을 위해 객체를 반환한다

hanurii committed:
[#33] Feat : delete 테스트로 삭제 테스트와 같은 포스트 중복 삭제 테스트

- delete 삭제를 여러 번 했을 때 오류가 나야한다.

hanurii committed:
[#33] Feat : api-guide.adoc 에 포스트 생성에 관한 문서 작성

- 여기에 포스트 생성에 관한 문서 작업을 해줘야지 Spring 을 실행했을 때 볼 수 있다

hanurii committed:
[#33] Modify : 필요없는 시간 필드 삭제

- 필요없는 시간 필드 삭제

hanurii committed:
[#33] Feat  : 포스트 생성 Sprng Rest Docs 문서 작성 (v 1.0)

- 포스트 생성 Spring rest docs 작성
- 아주 기본적인 내용만 들어있어서 내용 추가 예정

hanurii committed:
[#33] Modify : 필요없는 주석 삭제와 실패하는 테스트 주석 처리

- 실패하는 테스트가 있을 경우 Spring Rest Docs에서 오류가 난다
- 어떻게 하는지 아직 몰라서 주석으로 처리해둔다

102092 committed:
[#37] Feat : Account 추가

- 유저 엔티티
- 현재 id, avatarUrl 멤버변수만 가지고 있음

102092 committed:
[#37] Feat : Cors 오류

102092 committed:
[#37] Test : ApiDocumentUtils 추가

- ApiDocument의 기본 설정을 담당할 인터페이스 추가

hanurii committed:
Merge pull request #38 from Hydronium-ion/feature/issue-037

Feature/issue 037

102092 committed:
Merge pull request #39 from Hydronium-ion/dev

Author, ApiDocument

hanurii committed:
[#33] Feat : 포스트 삭제 Spring Rest Docs 완성(성공 케이스만)

- 포스트 삭제 Spring Rest Docs 성공 케이스만 완성했습니다
- 이후 실패 케이스도 만들 예정입니다

hanurii committed:
Merge branch 'dev' into feature/issue-033

hanurii committed:
[#33] Modify : Post 클래스의 author 의 타입을 String 에서 Account 객체로 바꿈

- 그러면서 생기는 수정할 것들을 수정

102092 committed:
[#38] Feat : createdTimeAt 변수명 수정

- createdTimeAt --> createdAt

102092 committed:
Merge pull request #42 from Hydronium-ion/feature/issue-038

[#38] Feat : createdTimeAt 변수명 수정

102092 committed:
Merge pull request #43 from Hydronium-ion/dev

CreatedTimeAt 변수명 변경

hanurii committed:
[#33] Feat : Account 객체 형태를 나타내기 위해서 소스 코드 수정

- Spring Rest Docs 로 객체를 나태내기 위해서

fieldWithPath("author").description("포스트 작성자").type(JsonFieldType.OBJECT),
fieldWithPath("author.id").description("포스트 작성자의 ID").type(JsonFieldType.NUMBER),
fieldWithPath("author.name").description("포스트 작성자의 이름").type(JsonFieldType.STRING),
fieldWithPath("author.avatarUrl").description("포스트 작성자의 아바타 URL").type(JsonFieldType.STRING)

형태로 작성

hanurii committed:
Merge branch 'dev' into feature/issue-033

hanurii committed:
Modify : createdTimeAt 을 createdAt으로 변경 (충돌 잘못 해결해서 다시 커밋)

102092 committed:
[#44] Feat : build, deploy 수정

- build archivesBaseName 설정
- 이에 따른 deploy sh 수정

hanurii committed:
Merge pull request #45 from Hydronium-ion/feature/issue-044

[#44] Feat : build, deploy 수정

102092 committed:
Merge pull request #47 from Hydronium-ion/dev

CD script 수정

hanurii committed:
[#33] Modify : PostRequestDto에서 필요없는 views, likes 삭제

- 처음 생성 될 때 어차피 0이기 때문에 따로 request 로 받지 않는다

hanurii committed:
[#33] Modify : 단순한 테스트 삭제

- 이미 Spring REST Docs로 테스트 작업을 진행했기 때문에 중복 작업이다.

hanurii committed:
[#33] Feat : 필요없는 필드 삭제와 andDo() 안에 필요한 메서드 삽입

- preprocessResponse/preprocessRequest 안에 prettyPrint() 값이 들어있었던 것을 전역 메서드로 선언한 getDocumentRequest/getDocumentResponse 으로 변경

102092 committed:
[#49] Feat : thumnail image 추가

102092 committed:
Merge pull request #50 from Hydronium-ion/feature/issue-049

[#49] Feat : thumnail image 추가

102092 committed:
Merge pull request #51 from Hydronium-ion/dev

썸네일 URL 랜덤으로

hanurii committed:
[#33] Feat : PostRequestDto의 필요없는 id 필드값을 지움으로써 변경된 사항에 따른 코드 수정

- 리퀘스트로 받는 id 값은 항상 null 일 것이기 때문에 id 필드값을 지웠다

hanurii committed:
[#33] Modify : 포스트 생성/삭제 시 응답 메세지 수정

- 한글 응답보단 영어 응답이 좋을 것 같다는 한님의 의견을 반영
- 영어를 잘했다면 더 적극적인 영어 응답을 썼을텐데 아쉽다

hanurii committed:
[#33] Feat : @ToString 삭제

- 엔티티에 @ToString 때문에 생길 수 있는 오류를 방지하고자 삭제

102092 committed:
Merge pull request #48 from Hydronium-ion/feature/issue-033

포스트 생성/삭제 기능 구현 및 Spring Rest Docs 문서 추가

102092 committed:
[#52] Feat : api-guide.adoc findById API 추가

102092 committed:
[#52] Refactor : import optimize

- 쓸 데 없는 의존성 삭제

102092 committed:
[#52] Feat : findById 테스트 추가

- 포스트 메서드 이름 수정, spring rest docs 이름에서 삭제
- findByID 테스트 추가

102092 committed:
[#52] Refactor : Post Create Request, Response 리팩토링

- 클래스명 수정. 좀 더 명료하도록
- CreateResponse 응답 객체 생성
- Post Entity 내부 메서드 수정
- controller 메서드명 수정 controller만 보고도, 내부 동작을 알 수 있도록

hanurii committed:
[#8] Feat : 좋아요 순으로 Content 정렬 구현

- 포스트의 좋아요 수를 기준으로 내림차순 정렬하여 출력합니다.

hanurii committed:
[#8] Feat : 테스트 코드 작성 및 Spring Rest Docs 작성

- 필수적인 것들만 문서화를 진행했다
- optional 인 부분들을 명시해주는 것은 차후 추가할 예정

102092 committed:
[#52] Refactor : code reformatting, 변수명 수정

- 가독성을 높히기 위해서, 코드 리포멧
- 사용 하지 않는 객체 삭제
- 이상한 변수명 수정

102092 committed:
[#52] Refactor : PostController 메서드 이름 수정

- 해당 메서드가 어떤 역할을 하고 있는지 정확하게 나타내기 위해서
- ...latest -> 최신순으로 모든 것을 찾아 주는 역할
- create -> Post를 생성하는 역할

hanurii committed:
Merge pull request #54 from Hydronium-ion/feature/issue-052

Feature/issue 052

102092 committed:
Merge pull request #55 from Hydronium-ion/dev

Issue-033, issue-052

hanurii committed:
[#8] Modify : 메서드명 변경

- 한님과 서로 합의한 메서드 명으로 변경합니다

hanurii committed:
Merge branch 'dev' into feature/issue-008

hanurii committed:
[#8] Modify : 컴파일 오류가 생긴 부분 수정

hanurii committed:
[#8] Modify : 테스트 코드 수정

- 코드 병합 시 postService가 없어졌기 때문에 postController로 바꿔서 mokito 테스트 진행

hanurii committed:
[#8] Modify : 테스트 코드 메서드명 변경에 따른 adoc 문서 내용 변경

hanurii committed:
[#8] Modify : Spring Rest Docs 이름의 통일성을 위해 변경

- Spring Rest Docs 이름의 통일성을 위해 변경

hanurii committed:
[#8] Modify : Integer -> int  타입으로 변경

- null 이 들어올 것을 예상했다면 랩퍼 클래스를 사용했을 것이다.
- 하지만 페이지와 사이즈는 클라이언트 단에서 숫자로 넣어줘야 하는 것이므로 int 형으로 변경

hanurii committed:
[#8] Modify : 테스트 @DisplayName 이름을 통일성있게 변경

- 가독성을 높이고자 통일성을 주었다

102092 committed:
Merge pull request #56 from Hydronium-ion/feature/issue-008

좋아요 순으로 Content 정렬(내림차순)

102092 committed:
[#58] Feat : PostCreateRequest 멤버 변수 추가

- subTitle, Public 여부를 판단하게 해주는 isPublic
- 필수적이라 생각되는 validation 추가

102092 committed:
[#58] Feat : Post Entity 변경

- PostCreateRequest 변경에 따른 엔티티 수정

102092 committed:
[#58] Feat : PostRepository 메서드 수정

- 매개변수로 Pageable을 받도록
- 기본적으로 IsPublic 이 True인 포스트만 반환하도록 수정함

102092 committed:
[#58] Feat : PostCreateRequestTest 작성

- Validation을 확인할 수 있는 테스트 작성

102092 committed:
[#58] Refactor : Post Entity 변경, PostCreateRequest 변경에 따른 테스트 수정

- 생성에 필요한 멤버 변수가 추가되어서, 기본적인 엔티티 생성 수정
- 반복되는 Post 생성을 메서드로 리팩토링
- 테스트 순서 변경, 조회 테스트가 상단으로 위치하도록

102092 committed:
[#58] Feat : PostController mapping 추가, 순서 변경

- posts/createdAt으로 수정, 명확하게 어떤 것을 기준으로 반환되는지 알기 위해서
- 메인 페이지 조회 기능이 해당 클래스 상단에 위치하도록

102092 committed:
[#58] Feat : Validation handler 추가

- Valid에서 발생하는 오류를 핸들하기 위한 핸들러를 추가하였음.

102092 committed:
[#58] Feat : findAllAndOrderByCreatedAtDesc 페이징 기능 추가

- 기본적인 페이징 기능을 추가함
- 향후 sorting 되는 기준을 enum등으로 관리해야할듯
- 기본적으로 공개된 포스팅만 반환하도록 수정

102092 committed:
[#58] Feat : 데이터 추가

- 어플리케이션 실행시, 데이터를 더 추가함
- Entity 변경에 따른, 생성에 필요한 변수 추가

102092 committed:
[#58] Feat : api-guide.adoc 순서 수정

- 메인 페이지 조회가 상단에 위치하도록

102092 committed:
[#58] Refactor : mvc perform에서 query 변경

102092 committed:
[#58] Refactor : 오타 수정, 공백 제거

hanurii committed:
Merge pull request #60 from Hydronium-ion/feature/issue-058

Feature/issue 058

hanurii committed:
[#69] Feat : Jasypt-spring-boot-starter 를 이용해 민감 정보 암호화 기본 세팅 구현

- 일단 로컬 DB와 관련된 정보를 암호화 했다

hanurii committed:
[#69] Feat : password는 중요하기 때문에 시스템 환경 변수로 설정하여 사용

- password가 그대로 노출되면 민감한 정보가 유출 될 위험이 있다

hanurii committed:
[#69] Feat : 배포되는 서버와 관련된 민감 정보들도 암호화

- 배포되는 서버와 관련된 민감 정보들도 암호화 진행
- 이번 프로젝트는 커밋이 남아있기 때문에 민감 정보를 깃헙 레포에서 삭제하려면 지난 커밋을 들춰서 삭제해야 한다
- 다음 번 프로젝트에는 프로젝트 세팅 때부터 세팅해놓고 시작하는 게 좋을 것 같다

hanurii committed:
[#69] Feat : Jsaypt로 암호화 되는지 테스트 코드 작성

hanurii committed:
Update CI-BE.yml

JASYPT_PASSWORD 환경 변수를 추가해주었습니다

hanurii committed:
[#69] Feat : Jasypt 테스트 클래스 생성 및 spring-datasource-url 옵션 추가

- Jasypt 테스트 클래스 생성 및 spring-datasource-url 옵션 추가

hanurii committed:
[#69] Feat : 클래스명을 명시해준다.

- 클래스명을 명시해준다.

hanurii committed:
Merge branch 'feature/issue-069' of https://github.com/Hydronium-ion/Rare-Server into feature/issue-069

hanurii committed:
[#69] Modify : 코드 리뷰에 관한 사항 수정 및 삭제

102092 committed:
Merge pull request #70 from Hydronium-ion/feature/issue-069

공개된 레파지토리에서 민감한 정보 숨기기

102092 committed:
Merge pull request #72 from Hydronium-ion/dev

8월 1주차

Hyune-c commented:
첫 PR 이라 그런가 크기가 어마어마하게 크군요. 열심히 봐보겠습니다.

Hyune-c commented:
아직 프로젝트 README.md 가 없나요? 
대충 들어서 알고는 있지만 README.md 에 내용이 존재해서 한번 보고 갔으면 좋겠습니다.

realsigridjin commented on build.gradle:
```diff
@@ -0,0 +1,61 @@
+    testImplementation('org.springframework.boot:spring-boot-starter-test') {
+        exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
+    }
+    testImplementation 'org.springframework.restdocs:spring-restdocs-mockmvc'
```
MockMvc보다는 RestAssured를 사용해보세요.

realsigridjin replied:
MockMvc말고 RestAssured 사용을 하면 다음과 같은 이점을 얻을 수 있다고 합니다.
* RestAssured를 사용하면, 예를 들어 Spring Rest Docs에서 mockmvc를 사용한다고 하더라도, 서블릿 컨테이너를 띄우되, RestAssured 문법을 사용할 수 있습니다.
* RestAssured라는 테스트 객체는 실제 서버가 띄워진 상태에서 테스트를 할 수 있는데요, mockmvc라는 것을 띄우면 실제 서버가 아닙니다. RestAssured의 경우 실제 말하는 서블릿 컨테이너가 mock 컨테이너를 띄우는 것이므로 가볍고, 세부 설정이 가능합니다.
* 그래서 mock 서블릿 컨테이너를 사용하는 restassured 사용을 추천 많이 하더라구요.

102092 replied:
좋은 라이브러리 추천 감사합니다.
이 점 까지 알려주시다니,, 꼭 써봐야겠네요 @Hamill210 🔥 !!!

Hyune-c commented on src/main/java/com/codesquad/rare/common/BaseTimeEntity.java:
```diff
@@ -0,0 +1,18 @@
+
+@Getter
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
```
`EntityListeners` 는 처음보는 것이라 검색해보니 `데이터 변경시 알림을 받는 방법으로 콜백을 이용` 하는 것인 것 같네요.
잘 모르는 내용이라 같이 공부해보겠습니다.

102092 replied:
저도 비슷하게 알고 있습니다. ㅎㅎ 엔티티에 어떤 변화가 있을 때, 그걸 캐치해서 어떤 작용을 하는 것으로 이해하고 있어요!
다만, 해당 클래스는 작성만 해놓고 아직 적용은 하지 않은 상태입니다.

102092 replied:
저도 비슷한 내용으로 이해하고 있습니다.
엔티티 변경을 잡아서, 어떤 행동을 하는 데 도와주는 어노테이션으로 알고있네요

realsigridjin commented on build.gradle:
```diff
@@ -0,0 +1,61 @@
+    testImplementation 'org.springframework.restdocs:spring-restdocs-mockmvc'
+    asciidoctor 'org.springframework.restdocs:spring-restdocs-asciidoctor'
+    testRuntimeOnly 'com.h2database:h2'
+    implementation 'com.github.gavlyukovskiy:p6spy-spring-boot-starter:1.5.6'
```
위 의존성을 추가하면 SQL 로그에서 파라미터에 전달되는 값까지 확인할 수 있습니다.
그런데 이는 성능에 영향을 미치므로 개발단계만 사용하고 운영단계에서는 제거하는 것을 추천합니다.

102092 replied:
알려주셔서 감사합니다.
SQL에 대해 자세히 알고 싶어서 추가했었는데, 성능상 문제가 있었군요..

하밀과 상의해보도록 하겠습니다! @Hamill210

Hyune-c commented:
- 이번 리뷰를 함에 있어서 아래의 약어를 사용할 예정입니다. 참고 부탁드립니다.
![image](https://user-images.githubusercontent.com/55722186/89610514-a8bf2200-d8b5-11ea-8179-688521be97ce.png)

realsigridjin commented on src/main/java/com/codesquad/rare/config/JasyptConfig.java:
```diff
@@ -0,0 +1,26 @@
+@Configuration
+public class JasyptConfig {
+
+  @Bean("jasyptStringEncryptor")
```
**도전 질문**
```@Configuration```과 ```@Bean``` 그리고 ```@Component```의 차이점에 대해 설명해보시오.

102092 replied:
검색하지 않고, 이 질문에 대답하는 건 어렵겠지만 아는 만큼 써보겠습니다.

`@Configuration` 이 붙은 클래스들은 스프링 시작될 때 먼저 컨테이너에 등록되고,
`@Bean`같은 경우, 미리 정의되어있는 클래스를 개발 환경에 맞춰서 다시 만들어서 등록 시킬 때?
`@Compoent` 는 ... 음. 개발자가 만든 클래스를 등록하려고 할때, 사용했던듯 싶어요.

realsigridjin commented on src/main/java/com/codesquad/rare/config/WebConfig.java:
```diff
@@ -0,0 +1,40 @@
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class WebConfig implements WebMvcConfigurer {
```
개인적으로 WebMvcConfigurer보다 WebMvcAutoConfiguration 사용을 추천한다.
[스택오버플로우 관련 링크](https://stackoverflow.com/questions/44280287/why-we-use-webmvcautoconfigurationadapter-class)

102092 replied:
이런 것도 있었군요!
그런데 어떤 면에서 전자보다 후자가 더 좋은 지 좀 더 알려주실 수 있나요?
링크에 적힌 답글만 봐서는 잘 모르겠어요..ㅠㅠ

realsigridjin commented on src/main/java/com/codesquad/rare/domain/Like.java:
```diff
@@ -0,0 +1,34 @@
+import org.hibernate.annotations.CreationTimestamp;
+
+@Getter
+@Builder
```
Builder 어노테이션을 클래스에 붙이는 것은 분명한 **안티패턴** 이라고 생각합니다.
그 이유는 다음과 같습니다.
- 클래스 위에 ```@Builder```를 사용 시 ```@AllArgsConstructor``` 어노테이션을 붙인 효과를 발생시켜 모든 멤버 필드에 대해서 매개변수를 받는 기본 생성자를 만듭니다. 이는 모든 멤버필드에 대한 매개변수를 허용하게 됩니다.
- 예를 들어 클래스의 id 생성전략은 데이터베이스의 auto_increment로 DB 상에서 주입되는 경우, 클래스 생성 시 id를 넘겨받지 않아야 합니다. createdAt, updatedAt 같은 경우는 ```@CreationTimestamp```, ```@UpdateTimestamp``` 각각의 어노테이션이 주입을 담당하고 있습니다. 클래스 위에 ```@Builder``` 어노테이션을 붙이는 경우 객체 생성시 받지 않아야 할 데이터들이 주입될 수 있습니다.
- 따라서 받아야 하는 생성자를 필요조건에 따라 지정하고 그 위에 ```@Builder```를 붙이는게 바람직하다고 생각합니다.

대안책으로, 생성자 하나를 만들어서 그 위에 ```@Builder``` 어노테이션을 지정해주는 것을 추천합니다.

[참고 링크](https://cheese10yun.github.io/lombok/)
[참고 링크 2](https://dev.to/siy/when-builder-is-anti-pattern-3j92)

102092 replied:
생각치도 못했던 부분이었습니다. 알려주셔서 감사합니다 :)

realsigridjin commented on src/main/java/com/codesquad/rare/domain/Like.java:
```diff
@@ -0,0 +1,34 @@
+public class Like {
+
+  @Id
+  @Column(name = "like_id")
```
그냥 Column 이름이 ```id```가 아닌 이유는 무엇일까용?

102092 replied:
김영환님 JPA 강의 책을 보다보니, 이렇게 하시더라구요..
아무래도 정확히 어떤 id인지 명시하기 위해서 그런듯 한데, 책을 보면서 하다보니 저희도 이렇게 적게 되었습니다

realsigridjin commented on src/main/java/com/codesquad/rare/domain/post/Post.java:
```diff
@@ -0,0 +1,86 @@
+
+  private LocalDateTime createdAt;
+
+  public static Post toEntity(PostCreateRequest postCreateRequest, Account author) {
```
정적 메소드 사용 👍

Hyune-c commented on src/main/java/com/codesquad/rare/common/api/ApiResult.java:
```diff
@@ -0,0 +1,30 @@
+@ToString
+public class ApiResult<T> {
+
+  private final boolean success;
```
Response Body 에 성공 여부에 대한 값이 꼭 필요할까요?
저는 Response Header 의 HTTP response status code 로 충분하다고 생각합니다.

102092 replied:
저도 그 부분에 대해 여러 고민을 해봤습니다.

프론트 하시는 분들과 이 부분에 대해 서로 약속을 한 건 아니지만,  

제가 느끼기에는 어떤 요청에 대한 결과를 Json으로 돌려주면, 앞에서(front) 시각적으로 보고 처리하기 편하지 않을까 라는 생각에 성공여부, status code등을 드리고 있습니다.

Hyune-c replied:
그렇다면 BE 에서는 1개의 결과를 주기위해 header 와 body 를 모두 만지는 것이 되는데,  
그런 경우 2가지의 문제가 있을 것으로 생각됩니다.

- header, body 의 값을 동일하게 유지하여야 됨.
  - 관리 포인트의 증가로 누락 위험.
- Body 의 크기 증가로 패킷 사이즈 증가.

위 2가지의 문제점과 FE 와 협의된 편의성 중 좋다고 생각하는 것을 선택하시면 되겟지만, 전 합치는 것을 추천드립니다.

realsigridjin replied:
맞아요. 너무 verbose 하다는 느낌을 지울 수 없긴 하네요 ㅎㅎ

realsigridjin commented on src/main/java/com/codesquad/rare/domain/post/PostDataRunner.java:
```diff
@@ -0,0 +1,114 @@
+@RequiredArgsConstructor
+@Slf4j
+@Component
+public class PostDataRunner implements ApplicationRunner {
```
ApplicationRunner 사용 👍 
추가적으로 CommandLineRunner를 학습해보세요.

102092 replied:
알려주셔서 감사합니다 :)

Hyune-c commented on src/main/java/com/codesquad/rare/common/api/ApiResult.java:
```diff
@@ -0,0 +1,30 @@
+  }
+
+  public static ApiResult ERROR(Exception exception, HttpStatus status) {
+    return new ApiResult<>(false, null, new ApiError(exception, status));
```
HttpStatus 가 Body 에 들어가면 Header 에 들어가는 값을 똑같이 또 보내는 것인가요?

102092 replied:
넵 맞습니다.

Hyune-c replied:
https://github.com/Hydronium-ion/Rare-Server/pull/78#discussion_r467133298

로 대신합니다.

realsigridjin commented on src/main/java/com/codesquad/rare/domain/post/PostDataRunner.java:
```diff
@@ -0,0 +1,114 @@
+
+    Random random = new Random();
+    LocalDateTime localDateTime = LocalDateTime.now();
+    List<Post> postList = new ArrayList<>();
```
```Lists.newArrayList```를 사용해보세요. 타입추론이 가능하답니다.

```java
List<Foo<Bar, Baz>> list = Lists.newArrayList();
List<Foo<Bar, Baz>> list = new ArrayList<Foo<Bar, Baz>>();
```

아래의 내용은 ```new ArrayList<>``` 에서는 불가능한 것입니다.
```java
Lists.newArrayList("a","b");
Lists.newArrayList(anIterable);
```

https://stackoverflow.com/questions/9980915/lists-newarraylist-vs-new-arraylist

102092 replied:
오호 이런것도 있었네요 신기해요ㅋㅋㅋ

Hyune-c commented on src/main/java/com/codesquad/rare/config/JasyptConfig.java:
```diff
@@ -0,0 +1,26 @@
+public class JasyptConfig {
+
+  @Bean("jasyptStringEncryptor")
+  public StringEncryptor stringEncryptor() {
```
저는 해보지 못했지만 암호화 멋있네요.

102092 replied:
하밀이 정말 많이 고생하셨어요... @Hamill210 👍

realsigridjin commented on src/main/java/com/codesquad/rare/domain/post/PostRepository.java:
```diff
@@ -0,0 +1,10 @@
+
+public interface PostRepository extends JpaRepository<Post, Long> {
+
+  Page<Post> findAllByIsPublicTrue(Pageable pageable);
```
정적 쿼리가 많아질 것 같은데 QueryDSL을 사용하여 작성해보세요.

102092 replied:
QueryDSL에 대해선 저도 잘 알지 못해서.. 혹시 추천해주실만한 블로그나 동영상이 있을까요?

Hyune-c replied:
김영한님 jpa 책 뒤쪽에 있어요.

realsigridjin commented on src/main/java/com/codesquad/rare/domain/post/PostService.java:
```diff
@@ -0,0 +1,58 @@
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
```
```@Transactional``` 클래스 선언 👍

102092 replied:
@Hamill210 의 센스있는 조언으로 비롯되었습니다

Hyune-c commented on src/main/java/com/codesquad/rare/common/BaseTimeEntity.java:
```diff
@@ -0,0 +1,18 @@
+@EntityListeners(AuditingEntityListener.class)
+public abstract class BaseTimeEntity {
+
+  @CreatedDate
+  private LocalDateTime createdTimeAt;
```
`LocalDateTime` 을 만듬에 있어서 `@CreatedDate` 으로 충분한가요? 
혹시 `@CreationTimestamp` 를 써야되지 않나요?

102092 replied:
@CreateDate는 Spring Data Jpa 기능이고,
@CreationTimestamp는 Hibernate 기능으로 알고있습니다.
- 참고 : https://stackoverflow.com/questions/49954812/how-can-you-make-a-created-at-column-generate-the-creation-date-time-automatical

Hyune-c replied:
그렇군요.
근데 그렇다면 두 어노테이션 중 한가지만 사용하는게 코드 전체에 있어 일관성을 가져가지 않을까요?

102092 replied:
맞습니다 .아직 이 부분에 대해 Entity를 적용하지 않아서, 추후 확실하게 하나를 선정하여 적용할 생각입니다

realsigridjin commented on src/main/java/com/codesquad/rare/domain/post/PostService.java:
```diff
@@ -0,0 +1,58 @@
+        .orElseThrow(() -> new NotFoundException(Account.class, postCreateRequest.getAuthorId()));
+    Post savedPost = postRepository.save(Post.toEntity(postCreateRequest, author));
+    PostCreateResponse response = new PostCreateResponse();
+    response.setPostId(savedPost.getId());
```
setter를 사용하지 마시고 PostCreateResponse의 인스턴스를 생성할 때 함께 파라미터로 넘기도록 리팩토링 해보세요.

102092 replied:
좋은 방향이네요.
리팩토링시 참고하도록 하겠습니다!

realsigridjin commented on src/test/java/com/codesquad/rare/document/utils/ApiDocumentUtils.java:
```diff
@@ -0,0 +1,25 @@
+import org.springframework.restdocs.operation.preprocess.OperationRequestPreprocessor;
+import org.springframework.restdocs.operation.preprocess.OperationResponsePreprocessor;
+
+public interface ApiDocumentUtils {
```
OperationRequestPreprocessor의 사용 👍

102092 replied:
우아한 형제들 블로그를 그대로 참고했습니다..

Hyune-c commented on src/main/java/com/codesquad/rare/domain/Like.java:
```diff
@@ -0,0 +1,34 @@
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@NoArgsConstructor
+@Entity
+@Table(name = "likes")
+public class Like {
```
굳이 디폴트가 아닌, 다르게 한 이유가 있을까요? 
잠재적인 위험이 생길 수 있다고 느껴집니다.

102092 replied:
Like가 예약어 더라구요... 그래서 부득이하게 수정하게 되었습니다..ㅠㅠ
- https://zetawiki.com/wiki/MySQL_%EC%98%88%EC%95%BD%EC%96%B4

Hyune-c replied:
아 그렇겟네..
근데 그렇다면 오히려 Entity 의 이름을 변경하는게 낫지 않을까요?

102092 replied:
좋아요는 어떤 이름이 좋을까요.. Favorite?

realsigridjin commented on src/test/java/com/codesquad/rare/domain/post/PostServiceTest.java:
```diff
@@ -0,0 +1,78 @@
+import org.springframework.restdocs.RestDocumentationExtension;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith({RestDocumentationExtension.class, SpringExtension.class})
```
```@ExtendWith```의 적절한 사용 👍 
테스트에서 모킹을 위해 SpringExtension.class 사용이 돋보이네요.

Hyune-c commented on src/main/java/com/codesquad/rare/domain/Tag.java:
```diff
@@ -0,0 +1,33 @@
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  //TODO name length 설정하기
```
column length 를 정하시면서 타입도 생각해보시면 좋을 것 같습니다.

102092 replied:
조언 감사합니다... 제약이 걸리는 만큼, 프론트분들하고도 이야기 나눠봐야할 부분 이네요...

realsigridjin commented on src/test/java/com/codesquad/rare/domain/post/PostServiceTest.java:
```diff
@@ -0,0 +1,78 @@
+   * 같은 포스트라도 반복적으로 ID 값만 다르다면 모두 생성할 수 있어야 합니다. 같은 포스트를 10번 반복 생성해보는 테스트입니다.
+   */
+  @DisplayName("같은 포스트 반복 생성 테스트")
+  @RepeatedTest(value = 10, name = "{currentRepetition}/{totalRepetitions} 번째 테스트")
```
JUnit 5의 풍부한 라이브러리 사용을 통한 가독성 증진 및 중복 제거가 우수합니다 👍

realsigridjin commented on src/test/java/com/codesquad/rare/domain/post/PostServiceTest.java:
```diff
@@ -0,0 +1,78 @@
+  @DisplayName("같은 포스트 반복 생성 테스트")
+  @RepeatedTest(value = 10, name = "{currentRepetition}/{totalRepetitions} 번째 테스트")
+  void create_tenTimes_AllPass(RepetitionInfo repetitionInfo) {
+    //given
```
given when then의 사용 👍

Hyune-c commented on src/main/java/com/codesquad/rare/domain/account/AccountRepository.java:
```diff
@@ -0,0 +1,7 @@
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface AccountRepository extends JpaRepository<Account, Long> {
+
+}
```
으악. Spring Data JPA..
전 세부 설정이 힘들어져서 포기했었습니다 ㅠㅠ

Hyune-c commented on src/main/java/com/codesquad/rare/domain/post/Post.java:
```diff
@@ -0,0 +1,86 @@
+@Builder
+@AllArgsConstructor(access = AccessLevel.PROTECTED)
+@NoArgsConstructor
+@JsonPropertyOrder({
+    "id",
+    "title",
+    "subTitle",
+    "content",
+    "thumbnail",
+    "author",
+    "views",
+    "likes",
+    "tags",
+    "createdAt"
+})
```
멋지네요.

하지만 이 클래스는 Entity 로 보이는데 왜 Json 에 대한 정의가 있는 걸까요?
Json 은 클라이언트로 내려갈 때 필요할 것으로 추정되는데, 그 경우 Entity 가 아닌 DTO 를 만들어서 내려보내야 된다고 생각합니다.

realsigridjin replied:
저도 동의합니다. 이 부분은 별도의 DTO로 구현되어야 합니다.

102092 replied:
맞습니다!!
저도 Entity 그대로 내리는 것에 대한 위험을 알고 있지만,
아무래도 응답 객체들에 대한 설계가 늦어져서, 위와 같이 설정하고 내려주고 있었네요..ㅠㅠ
추후 JsonPropertyOrder는 entity에서 사라질 예정입니다!

Hyune-c commented on src/main/java/com/codesquad/rare/domain/post/Post.java:
```diff
@@ -0,0 +1,86 @@
+
+  private LocalDateTime createdAt;
+
+  public static Post toEntity(PostCreateRequest postCreateRequest, Account author) {
```
정적 팩토리 메소드를 만듬에 있어서 `of` `from` 이 아닌 `toEntity` 라는 이름을 한 이유가 있을까요?

일단 일반적이지 않고 메소드의 이름이 길다고 생각합니다.

realsigridjin replied:
보통 ```toEntity```라는 표현은 DTO에서 Entity를 생성할 때 나오는 표현입니다.
제가 리팩토링한다면 정적 팩토리 메소드를 만들 때 ```ofRequestAndAccount``` 정도로 정할 것 같습니다.

102092 replied:
메서드 이름이 잘못되어있었네요. 리팩토링 부분에서 수정해야겠습니다! 알려주셔서 감사해요

Hyune-c replied:
전 그냥 `of` 라고만 할거 같아요. 만약 인자 값이 복수개의 객체라는 점이 걸린다면 `from` 까지는 괜찮을거 같아요.
하지만 개인적으로 거의 모든 경우에 `of` 로만 쓰는 것을 선호하긴 합니다.

https://blog.javarouka.me/2018/11/12/static-creator/

- from
  - 매개변수 하나인 메서드
- of
  - 매개변수가 N개인 메서드

realsigridjin reviewed (approved):
전반적으로 구현 잘 하셨습니다.
필요한 스프링 라이브러리와 테스트를 위한 JUnit 5 라이브러리의 풍부한 사용이 인상적이네요.
커밋 로그를 확인해보니깐 TDD에 준하는 개발 경험을 이어가고 계시는 것 같아 매우 좋았습니다.
저는 이번 기회에 ATDD 인수 테스트 경험을 통해서 하나의 API를 만들어가는 경험을 해보시는 것을 추천드립니다.
다음 리뷰를 기대하겠습니다 👍

Hyune-c commented on src/main/java/com/codesquad/rare/domain/post/PostController.java:
```diff
@@ -0,0 +1,63 @@
+  public ApiResult<List<Post>> findAllInLatestOrder(
+      @RequestParam(value = "page", required = false, defaultValue = DEFAULT_PAGE) int page,
+      @RequestParam(value = "size", required = false, defaultValue = DEFAULT_SIZE) int size) {
+    return OK(postService.findAllAndOrderByCreatedAtDesc(page, size));
```
Service 단의 메소드 이름이 모두 너무 길다고 생각합니다.
아마 Repository 의 것을 그대로 차용한 것으로 보이는데.. 기능이 많아질수록 가독성이 떨어진다고 생각합니다.

Service 단의 메소드를 좀 더 단순화 시키고 파라미터를 받는 형태를 추천합니다.
아래는 하나의 예시입니다.

```suggestion
    return OK(postService.findAll(page, size, reverseOrder));
```

102092 replied:
좋은 방향이네요!
차후 리팩토링 진행시, 서비스 메서드를 좀 더 깔끔하게, 그리고 파라미터를 통해 결과값이 조절되도록 만들어보겠습니다 
감사합니당 ㅎㅎ

realsigridjin commented:
![image](https://user-images.githubusercontent.com/41055141/89635022-4e3bbb00-d8e1-11ea-82c1-9525122d6481.png)

#### 새로운 방법으로 TDD를 접근
When we’re implementing a feature, we start by writing an acceptance test, which exercises the functionality we want to build. While it’s failing, an acceptance test demonstrates that the system does not yet implement that feature; when it passes, we’re done.

Growing object-oriented software, guided by tests

#### 새로운 방법으로 TDD를 접근
우리는 기능을 구현할 때, 만들고자 하는 기능을 수행하는 인수 테스트를 작성하는 것으로 시작한다. 인수 테스트가 실패하는 동안은 시스템이 아직 그 기능을 구현하지 않고 있다는 것을 보여준다; 인수 테스트가 통과되면, 기능 구현은 끝이다.

테스트 주도 개발로 배우는 객체 지향 설계와 실천

Hyune-c commented on src/test/java/com/codesquad/rare/domain/post/request/PostCreateRequestTest.java:
```diff
@@ -0,0 +1,66 @@
+
+class PostCreateRequestTest {
+
+  final Logger log = LoggerFactory.getLogger(PostCreateRequestTest.class);
+
+  Validator validator;
```
접근 한정자 누락!

102092 replied:
하밀과 이야기를 나눈적이 있지만, 저 같은 경우 서비스가 되는 코드가 아니니, 넣지 않아도 되질 않을까 + 참고하는 블로그에서도 안 넣은 경우들이 많더라구요..

@Hyune-c 는 어떻게 생각하시나요?

Hyune-c replied:
넣지 않으면 `default` 로 지정 될텐데, 솔직히 그렇게 써본 적이 없어서 어떤 사이드이펙트가 있을지 잘 모르겠네요..
저는 항상 붙이고 있어요.

Hyune-c commented on src/test/java/com/codesquad/rare/domain/post/PostServiceTest.java:
```diff
@@ -0,0 +1,78 @@
+@ExtendWith({RestDocumentationExtension.class, SpringExtension.class})
+class PostServiceTest {
+
+  final Logger log = LoggerFactory.getLogger(PostServiceTest.class);
```
접근 한정자 누락!

realsigridjin commented:
![image](https://user-images.githubusercontent.com/41055141/89635199-93f88380-d8e1-11ea-950f-0d6d3ba1102b.png)
다음의 과정을 지켜보면서 ATDD 개발경험을 가져보는 것을 추천드립니다.

[참고 링크](https://boorownie.github.io/2019-11-27/atdd_spring_boot)

Hyune-c commented on src/main/java/com/codesquad/rare/error/exeception/NotFoundException.java:
```diff
@@ -0,0 +1,8 @@
+public class NotFoundException extends ServiceRuntimeException {
+
+  public NotFoundException(Class cls, Object values) {
+    super("ID: "+values + ", " + cls.getSimpleName() + "을(를) 찾을 수 없습니다.");
```
String 의 `+` 연산은 좋지 않습니다.
심지어 `+` 사이의 스페이스도 맞지 않네요 😱

realsigridjin replied:
StringBuilder와 String의 차이점에 대해 학습해보세요.
[참고 링크](https://novemberde.github.io/2017/04/15/String_0.html)

102092 replied:
ㅎㅎ 간단하게 처리하려다 보니...
에러 처리는 항상 고민입니다. 어떻게 해야 에러 발생 시, 읽을만한 에러를 앞으로 보내줄 수 있을까요..

Hyune-c replied:
아직 개발 초기단계라고 생각합니다.
이 단계에서는 적절한 에러 메세지보다는 에러 객체의 트레이스 로그를 통으로 내리는걸 선호합니다.
그리고 이후 기능 개발이 되고 본격적으로 에러에 집중할 때 커스텀 메세지를 만드는 편입니다.

제가 만든 표준 에러 객체 샘플을 첨부합니다.

```java
@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class ExceptionResponse {

  private final String exceptionClass;
  private final String message;

  public static ExceptionResponse from(Exception e) {
    return ExceptionResponse.builder()
        .exceptionClass(e.getClass().toString())
        .message(e.getMessage())
        .build();
  }
}
```

Hyune-c commented on src/main/java/com/codesquad/rare/error/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,51 @@
+    log.error("MethodArgumentNotValidException occurred: {}", e.getMessage(), e);
+    return newResponse(e, HttpStatus.BAD_REQUEST);
+  }
+
```
개행!
```suggestion
```

102092 replied:
헉 감사합니다!

Hyune-c commented on src/main/java/com/codesquad/rare/domain/post/response/PostCreateResponse.java:
```diff
@@ -0,0 +1,16 @@
+public class PostCreateResponse {
+
+  private Long postId;
+
```
개행!
```suggestion
```

Hyune-c commented on src/main/java/com/codesquad/rare/domain/post/PostService.java:
```diff
@@ -0,0 +1,58 @@
+
+  public List<Post> findAllByLikesInDescendingOrder(int page, int size) {
+    PageRequest pageRequest = PageRequest.of(page, size, Sort.by("likes").descending());
+    return postRepository.findAllByIsPublicTrue(pageRequest).getContent();
```
이 메소드가 무엇을 하고 무엇을 리턴하는지 조금 더 명확해지도록 개행을 하면 좋을 것 같습니다.
```suggestion
    return postRepository.findAllByIsPublicTrue(pageRequest)
                          .getContent();
```

102092 replied:
좋은 방향이네요 ㅎㅎ
@Hamill210 은 개행을 저런 방식으로 하시는 것에 대해 괜찮으신가요?

Hyune-c reviewed (commented):
- 전체적으로 아직 DTO 에 대한 고려가 덜 되어 있다는 느낌이 들었습니다.  
아마 초기라 그런 것 같기도 하구요. 다음 버전에서는 좀 더 개선될 것이라고 생각합니다.
- 서비스단의 메소드명이 너무 길고 레포지토리의 것과 같다고 느꼇습니다.  
좀 더 개선할 수 있다고 생각합니다.

양이 많다보니 큰 것 위주로 코멘트로 남겼습니다.
아마 다음 리뷰에는 조금 더 세세하게 갈 거 같아요.

수고하셨습니다.

102092 committed:
[#79] Feat : method, origin 명시

- cors 설정에, 모든 것을 열지 않고 꼭 필요한 부분만 열도록 명시

102092 committed:
Merge pull request #80 from Hydronium-ion/fix

[#79] Feat : method, origin 명시

hanurii committed:
[#73] Modify : 배포 시 로그 확인을 위해 코드 수정

- CodeDeploy 시 배포가 되지 않는 현상을 위해 코드 수정

hanurii committed:
Merge pull request #83 from Hydronium-ion/feature/issue-073

[#73] Modify : 배포 시 로그 확인을 위해 코드 수정

hanurii committed:
Merge pull request #84 from Hydronium-ion/dev

CodeDeploy 버그 수정을 위해 쉘스크립트 수정

hanurii committed:
[#73] Feat : 배포 스크립트와 appspec 에 beforeInstall 추가

hanurii committed:
Merge pull request #85 from Hydronium-ion/feature/issue-073

배포 스크립트와 appspec 에 beforeInstall 추가

hanurii committed:
Merge pull request #86 from Hydronium-ion/dev

배포 스크립트와 appspec 에 beforeInstall 추가

hanurii committed:
[#73] Modify : 띄어쓰기

hanurii committed:
[#73] Feat : jasypt 버전 변경과 jasyptConfig 변경

hanurii committed:
[#73] Mofidy : Bean 을 다른 걸로 바꿔 끼워줌

hanurii committed:
Merge pull request #87 from Hydronium-ion/feature/issue-073

Jasypt 암호화로 인해 돌아가지 않는 인프라 수정

hanurii committed:
Merge pull request #88 from Hydronium-ion/dev

Jasypt 암호화로 인해 돌아가지 않는 인프라 수정

hanurii committed:
[#73] Modify : timeout 시간을 늘림

hanurii committed:
[#73] Modify : 관리자 계정인 root에서 ubuntu로 runas 설정 변경

hanurii committed:
Merge pull request #89 from Hydronium-ion/feature/issue-073

appspec.yml 변경

hanurii committed:
Merge pull request #90 from Hydronium-ion/dev

appspec.yml 수정

hanurii committed:
[#73] Modify : runas 권한 변경

hanurii committed:
Merge pull request #91 from Hydronium-ion/feature/issue-073

runas 권한 변경

hanurii committed:
Merge pull request #92 from Hydronium-ion/dev

runas 권한 변경

hanurii committed:
[#73] Modify : Jasypt 문제로 관련 코드 삭제. 그리고 암호화를 위해 암호화가 필요한 것들 환경변수로 설정

- Jasypt가 문제가 있어서 CodeDeploy 시 배포가 되지 않았던 것
- appspec.yml을 이전 코드로 변경
- build.gradle 에서 jasypt 의존성 제거
- deploy.sh 에서 자바 애플리케이션 로그를 보기 위해 스크립트 수정
- application.yml 암호화가 필요한 코드들을 환경변수로 설정

hanurii committed:
Merge pull request #93 from Hydronium-ion/feature/issue-073

Modify : Jasypt 문제로 관련 코드 삭제. 그리고 암호화를 위해 암호화가 필요한 것들 환경변수로 설정

hanurii committed:
Merge pull request #94 from Hydronium-ion/dev

Modify : Jasypt 문제로 관련 코드 삭제. 그리고 암호화를 위해 암호화가 필요한 것들 환경변수로 설정
